modules/baremetal/main.tf

```diff
@@ -0,0 +1,38 @@
+resource "ibm_is_bare_metal_server" "bms" {
+  profile        = var.profile
+  name           = var.prefix
+  image          = var.image
+  zone           = var.zone
+  keys           = var.ssh_key_ids
+  vpc            = var.vpc_id
+  bandwidth      = var.bandwidth
+  access_tags    = var.access_tags
+  resource_group = var.resource_group_id
+
+  # Attach subnet if VLANs are **not** provided
+  dynamic "primary_network_interface" {
```

Comment:
Creating a primary like this is considered legacy now, the new recommendation is to use a VNI attachement instead.

Is there a reason why you only use attached VNI only when using VLANs? Can you not use an attached VNI in either case?